### PR TITLE
Fix 2635

### PR DESCRIPTION
**Pull request by antgonza**
First review/merge https://github.com/biocore/qiita/pull/2638

This also solves: https://github.com/biocore/qiita/issues/2552 and improves the speed of the listing which was getting a bit too much.

Before:
```
In [10]: %timeit _build_study_info(User('antoniog@ucsd.edu'), 'public')
1 loop, best of 3: 2.25 s per loop

In [11]: %timeit _build_study_info(User('antoniog@ucsd.edu'), 'user')
1 loop, best of 3: 8.6 s per loop
```

Now:
```
In [19]: %timeit generate_study_list(User('antoniog@ucsd.edu'), 'public')
1 loop, best of 3: 1.94 s per loop

In [20]: %timeit generate_study_list(User('antoniog@ucsd.edu'), 'user')
1 loop, best of 3: 6.41 s per loop
```

### CODE REVIEW

File: qiita_db/test/test_util.py [hunk 1]
```diff
@@ -868,46 +873,134 @@ def test_generate_study_list(self):
...
+            'owner': 'Shared', 'shared': [],
+            'study_abstract': 'Some abstract goes here',
+            'pi': ('lab_dude@foo.bar', 'LabDude'), 'publication_doi': [],
+            'study_alias': 'TST', 'study_tags': None,
+            'number_samples_collected': 0}
+        exp1 = [sinfo]
+        exp2 = [snew_info]
+        exp_both = [sinfo, snew_info]
+
+        # let's make sure that everything is private for stiudy 1
```
Comment (ElDeveloper) [inline qiita_db/test/test_util.py]: stiudy -> study

File: qiita_db/util.py [hunk 2]
```diff
@@ -1366,78 +1368,97 @@ def generate_study_list(study_ids):
...
+        if user.level == 'admin':
+            sids = (sids |
+                    qdb.study.Study.get_ids_by_status('sandbox') |
+                    qdb.study.Study.get_ids_by_status('private') |
+                    qdb.study.Study.get_ids_by_status('awaiting_approval'))
+    elif visibility == 'public':
+        sids = qdb.study.Study.get_ids_by_status('public') - sids
+        visibility_sql = "visibility = 'public' AND"
+    else:
+        raise ValueError('Not a valida visibility: %s' % visibility)
```
Comment (ElDeveloper) [inline qiita_db/util.py]: valida -> valid

File: qiita_db/util.py [hunk 3]
```diff
@@ -1366,78 +1368,97 @@ def generate_study_list(study_ids):
...
-            info['shared'] = []
-            if info['shared_with_name'] and info['shared_with_email']:
-                for name, email in zip(info['shared_with_name'],
-                                       info['shared_with_email']):
-                    if not name:
-                        name = email
-                    info['shared'].append((email, name))
-            del info["shared_with_name"]
-            del info["shared_with_email"]
+    sids = []
```
Comment (ElDeveloper) [inline qiita_db/util.py]: This line is not needed, the code in 1374 should return an empty set if there are no studies at all.

Just tried something similar in the CLI and it works as expected:

```python
>>> x = set(i for i in [])
>>> x
set([])
```

Comment (antgonza) [reply qiita_db/util.py]: Thanks, in fact this is leftover from code transition ...

**Comment by ElDeveloper** [comment]
Thanks @antgonza !

**Pull request merged**